server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -2416,6 +2418,29 @@ private static void ensureNoMoreSelectorSeparators(String remainingExpression, S
                 );
             }
         }
+
+        /**
+         * Checks the expression for cross-cluster syntax and throws an exception if it is combined with ::failures selector.
+         * @throws IllegalArgumentException if cross-cluster syntax is detected after parsing the selector expression
+         */
+        private static void ensureNoCrossClusterExpressionWithFailuresSelector(
+            String expressionWithoutSelector,
+            IndexComponentSelector selector,
+            String originalExpression
+        ) {
+            if (selector == IndexComponentSelector.FAILURES) {
```

Comment:
Should we simply fully bar any selector usage with cross cluster search? If the only supported option is `::data`, and no selector is the same as that, perhaps we should just disallow the syntax in case the suffix of `::data` causes issues further down the line.

Reply:
I originally suggested only restricting `::failures` but on second thought restricting `::data` makes sense -- I guess it makes things simpler to just not allow any selectors at all.

Reply:
Thanks for raising this. If there is no real need to support `::data` it would make this more consistent and simpler. I'll make the change.